sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py

```diff
@@ -393,13 +407,15 @@ async def set_share_properties(self, **kwargs):
         timeout = kwargs.pop('timeout', None)
         access_tier = kwargs.pop('access_tier', None)
         quota = kwargs.pop('quota', None)
+        root_squash = kwargs.pop('root_squash', None)
```

Comment:
you don't have to pop, root_squash will be passed into set_properties() through kwargs on line 420